resources/services/iam/roles_fetch.go

```diff
@@ -20,7 +21,23 @@ func fetchIamRoles(ctx context.Context, meta schema.ClientMeta, parent *schema.R
 		if err != nil {
 			return err
 		}
-		res <- response.Roles
+		wrappedRoles := make([]*ohaws.WrappedRole, len(response.Roles))
+		for i, role := range response.Roles {
+			wrappedRoles[i] = &ohaws.WrappedRole{
```

Comment:
*nit

you can set the embedded struct directly with something like

```
ohaws.WrappedRole{
  Role: role,
}
```

[example](https://go.dev/play/p/mwfxxWTxJsj)

Reply:
Will do, thanks!